qt/scientific_interfaces/Muon/ALCInterface.cpp

```diff
@@ -105,16 +104,16 @@ void ALCInterface::initLayout() {
   m_dataLoading->initialize();
   m_dataLoading->setParent(this);
 
-  m_baselineModellingView = new ALCBaselineModellingView(m_ui.baselineModellingView);
-  m_baselineModelling = new ALCBaselineModellingPresenter(m_baselineModellingView, m_baselineModellingModel);
+  auto baselineModellingModel = std::make_unique<ALCBaselineModellingModel>();
+  ALCBaselineModellingView *baselineModellingView = new ALCBaselineModellingView(m_ui.baselineModellingView);
+  m_baselineModelling = new ALCBaselineModellingPresenter(baselineModellingView, std::move(baselineModellingModel));
```

Comment:
The old code seems to use an owning raw pointer for the presenter! It would be great to use a `std::unique_ptr` for the presenter so we do not need to worry about managing its lifetime